woocommerce/woocommerce-ios
# REST API: Migrate Stripe and WCPay endpoints

Closes: #8648 


## Description

This PR enables WCPay and Stripe endpoints to use REST API and application password authentication. Related mappers are also updated to parse contents without the data envelope.

Also: `WordPressSite` was updated to parse `gmtOffset` as either a number or a string. The default wp-admin URL was also updated to have a "/" at the end to fix the issue opening the page for finishing payment setup.

## Testing instructions

Pre-requisite: Make sure that you have a self-hosted store set up for IPP with both WCPay and Stripe. Setup guides can be found in PdfdoF-D-p2 and PdfdoF-oU-p2. You'll need a card reader to test this.
- Enable the feature flag `applicationPasswordAuthenticationForSiteCredentialLogin` and build the app to a physical device.
- On the prologue screen, select Enter your site address and proceed with the address of your test store.
- Log in with your site credentials. After the login succeeds, you should be navigated to the home screen.
- Navigate to the Menu tab and select Payments. Select WCPay as the payment option.
- Select Collect Payment and proceed with an IPP. Your device should be able to connect to your card reader and complete the payment.
- On the Payments screen, switch the Payment Provider to WC Stripe Gateway. Repeat the above step to make sure that the payment also succeed.
- Create an order on your test store with cash on delivery.
- Open the order on the app and select Collect Payment to proceed with IPP.
- After the payment succeeds, select Issue Refund on the order detail screen. Select an item and tap Next. In the Refund via row, the correct payment method should be displayed.

## Screenshots

N/A

---
